### PR TITLE
enable cursor on westeros compositor

### PR DESCRIPTION
**Pull request by moorthy-bs**
- introduced new field in json config for cursor file
    - enabling cursor only if any file mentioned in the json config
    - reading PNG image in ARGB888 format as westeros expects

Signed-off-by: Moorthy Baskar <moorthy.baskaravenkatraman-sambamoorthy@linaro.org>

### CODE REVIEW

**Comment by moorthy-bs** [comment]
Mouse event handling yet to upstream. implementation and testing is under progress

File: Compositor/CMakeLists.txt [hunk 1]
```diff
@@ -22,6 +22,7 @@ set(VERSION_REVISION 0)
 
 set(VERSION ${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_REVISION})
 
+find_package(PNG REQUIRED)
```
Comment (pwielders) [inline Compositor/CMakeLists.txt]: As the cursor is only available in WesterOS suggest to move this dendency to the proper place, begin: Compositor/lib/Wayland/CMakeLists.txt and not in this CMakeList.txt as now aslo the RPI and Nexus compositorrequire PNG but they do not have any curso Support

File: Compositor/Compositor.config [hunk 2]
```diff
@@ -119,6 +119,7 @@ map()
 
     endif ()
 
+    kv(cursor ${PLUGIN_COMPOSITOR_CURSOR_FILE})
```
Comment (pwielders) [inline Compositor/Compositor.config]: I guess thhis needs to be condition, only if ${PLUGIN_COMPOSITOR_CURSOR_FILE} != "" we add this keyword.

File: Compositor/lib/Wayland/Westeros.cpp [hunk 3]
```diff
@@ -161,9 +165,113 @@ namespace Westeros {
         {
         }
 
+        static unsigned char* readPNGFile(const char *filename, unsigned char &in_width, unsigned char &in_height)
```
Comment (pwielders) [inline Compositor/lib/Wayland/Westeros.cpp]: All our method start with a capital, so could we make this ReadPNGFile ?

File: Compositor/Compositor.config (hunk 2 above)
Comment (moorthy-bs) [reply Compositor/Compositor.config]: > I guess thhis needs to be condition, only if ${PLUGIN_COMPOSITOR_CURSOR_FILE} != "" we add this keyword.

okay

File: Compositor/lib/Wayland/Westeros.cpp (hunk 3 above)
Comment (moorthy-bs) [reply Compositor/lib/Wayland/Westeros.cpp]: > All our method start with a capital, so could we make this ReadPNGFile ?

okay

File: Compositor/CMakeLists.txt (hunk 1 above)
Comment (moorthy-bs) [reply Compositor/CMakeLists.txt]: > As the cursor is only available in WesterOS suggest to move this dendency to the proper place, begin: Compositor/lib/Wayland/CMakeLists.txt and not in this CMakeList.txt as now aslo the RPI and Nexus compositorrequire PNG but they do not have any curso Support

Thanks for this finding. Let me update it.

**Comment by pwielders** [comment]
And again a new feature from the community! Thank you !

**Pull request merged**